Sources/Basic/TerminalController.swift

```diff
@@ -85,9 +85,11 @@ public final class TerminalController {
 
         // Try determining using ioctl.
         var ws = winsize()
+#if !arch(powerpc64le)
```

Comment:
Can you add a comment that this doesn't work on ppc. We also also include the above `ws` variable in this #if statement.

Reply:
@aciidb0mb3r I understand that TIOCGWINSZ is supported on ppc64le and I guess "sys/ioctl.h" should be included to get that definition. I would like to give it a try to see if that actually works on ppc64le. However, how should this file be included in the swift code? I see "import" statements at the beginning, so is "import sys/ioctl.h" correct swift statement? Sorry, I am not a swift developer, hence don't know much about it. Could you please help me with swift code?

Reply:
Swift uses Clang modulemaps for importing C headers. For system headers, there is a modulemap shipped with macOS and a modulemap in the Swift repository for Linux. I am not really sure if Swift has a modulemap for ppc. /cc @jrose-apple

Reply:
Also, the official Swift language book is an excellent resource for learning the language! The book is freely available as a digital copy. Here is the link: https://itunes.apple.com/us/book/the-swift-programming-language-swift-4-0-3/id881256329?mt=11

Reply:
Module maps are by OS, not by architecture. I assume the ppc64le we're trying to bring up is a Linux, right?

Reply:
Yes, it is Linux.

Reply:
It looks like we've manually mapped [a number of the TIOC* constants](https://github.com/apple/swift/blob/master/stdlib/public/Platform/TiocConstants.swift) over in the "Glibc" module (already imported in this file), probably because they are macros that don't import well to Swift. (The rest of ioctl is probably already present.) But right now, the constants are limited to architectures that we know have them. Maybe Linux should be added to that list, at least for some of them?

This would involve figuring out which ones do not import on Linux, and whether the values listed are correct.